src/main/java/seedu/duke/command/Command.java

```diff
@@ -86,6 +89,69 @@ public void execute(RecipeList recipeList, UI ui) throws IOException {
                 ui.showAddingRecipeErrorMessage(e);
             }
             break;
+        case ADDTORECIPE:
+            try {
+                if (fullDescription.isEmpty()) {
+                    throw new IncompleteInputException("The description of " + type + " cannot be empty.\n");
+                }
+                if (!Parser.isValidAddToRecipe(fullDescription)) {
+                    ui.showInvalidAddToRecipeDescription();
+                    break;
+                }
+                String[] parsed = Parser.parseAddToRecipeDescription(fullDescription);
+                if (parsed.length != 3) {
+                    ui.showInvalidAddToRecipeDescription();
+                    break;
+                }
+                String elementType = parsed[0];
+                String id = parsed[1];
+                String description = parsed[2];
+                if (description.trim().equals(StringLib.EMPTY_STRING) && elementType.equals("step")) {
```

Comment:
In such a case, simply trimming only when comparing may result in the description not being trimmed when you pass it into the `Parser#isDuplicateIngredient(IngredientList, String)` and `Parser#isDuplicateStep(StepList, String)` method to check through the existing ingredients/steps.

Reply:
Trimmed in parseAddToRecipeDescription/parseDeleteFromRecipeDescription. Mark as resolved.